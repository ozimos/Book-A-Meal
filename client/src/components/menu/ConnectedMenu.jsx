import React from 'react';
import ReactModal from 'react-modal';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Accordion,
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from 'react-accessible-accordion';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import SearchInput, { createFilter } from 'react-search-input';
import MealDisplayCard from '../mealCard/MealDisplayCard';
import MealCardContainer from '../mealCard/MealCardContainer';
import Greeting from '../greeting/Greeting';
import { mealActions, menuActions } from '../../redux/actions';
import ConnectedNav from '../nav/ConnectedNav';
import '../../../public/styles/bookameal.scss';
import '../../../public/styles/modalOpenButton.scss';
import '../../../public/styles/search-input.css';
import '../../../public/styles/accordion.css';

ReactModal.setAppElement(document.getElementById('root'));
class Menu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      menu: [],
      dbMenu: []
    };
    this.addToMenu = this.addToMenu.bind(this);
    this.postMenu = this.postMenu.bind(this);
  }
  static getDerivedStateFromProps(nextProps, prevState) {
    let incomingMenu = [];
    let prevMenu = [];
    if (prevState.dbMenu) {
      prevMenu = prevState.dbMenu.map(meal => meal.id);
    }
    if (nextProps.menu) {
      incomingMenu = nextProps.menu.map(meal => meal.id);
    }
    const incomingMenuSet = new Set(incomingMenu);
    const difference = prevMenu.filter(x => !incomingMenuSet.has(x));
    if (difference.length > 0) {
      const newMenu = [...nextProps.menu];
      newMenu
        .forEach((meal) => { delete meal.MealMenus; });
      return {
        dbMenu: nextProps.menu,
        menu: newMenu };
    }
    return null;
  }
  async componentDidMount() {
    this.props.dispatch(mealActions.getAllMeals());
    await this.props.dispatch(menuActions.getMenu());
    const newMenu = [...this.props.menu];
    newMenu
      .forEach((meal) => { delete meal.MealMenus; });
    // eslint-disable-next-line
     this.setState({ menu: newMenu, dbMenu: this.props.menu });
  }

  addToMenu(meal) {
    const inMenu = this.state.menu.some(elem => elem.id === meal.id);
    if (!inMenu) {
      this.setState(prevState => ({ menu: [...prevState.menu, meal] }));
      toast.success('Meal has been added to menu', { className: 'toaster' });
    } else {
      toast.error('Meal is already in menu', { className: 'toaster' });
    }
  }

  removeFromMenu = id =>
    this.setState(prevState =>
      ({ menu: prevState.menu.filter(elem => elem.id !== id) }));
  searchUpdated = (term) => {
    this.setState({ searchTerm: term });
  }
  postMenu() {
    const mealIdArray = this.state.menu.map(meal => meal.id);
    this.props.dispatch(menuActions.postMenu({ meals: mealIdArray }));
    toast.success('Menu for the day has been posted', { className: 'toaster' });
  }

  render() {
    const KEYS_TO_FILTERS = ['id', 'title', 'description', 'price'];

    const filteredMeals = this.props.meals
      .filter(createFilter(this.state.searchTerm, KEYS_TO_FILTERS));
    const { isCaterer, firstName } = this.props.user;
    const isMenuSet = this.state.menu.length !== 0;

    return (
      <div className="contain">
        <header className="header">
          <ConnectedNav />
        </header>
        <main>
          <ToastContainer autoClose={2000} />

          <Greeting isCaterer={isCaterer} firstName={firstName} />
          <Accordion accordion>
            <AccordionItem expanded>
              <AccordionItemTitle>
                <div className="title-element flexbox">
                  <h5>
                  Your Meals
                  </h5>
                  <div
                    className="accordion__arrow u-position-relative"
                    role="presentation"
                  />
                </div>
              </AccordionItemTitle>
              <AccordionItemBody>
                <SearchInput
                  className="search-input input-field"
                  onChange={this.searchUpdated}
                />
                <MealCardContainer
                  meals={filteredMeals}
                  MealCard={MealDisplayCard}
                  collection="Menu"
                  addToCollection={this.addToMenu}
                  addClass="scroll"
                />
              </AccordionItemBody>
            </AccordionItem>
            <AccordionItem>
              <AccordionItemTitle>
                <div className="title-element flexbox wrap">
                  <h5>
          Today&#39;s Menu
                  </h5>
                  <div
                    className="accordion__arrow u-position-relative"
                    role="presentation"
                  />
                </div>
              </AccordionItemTitle>
              <AccordionItemBody>
                <div className="title-element flexbox">
                  <button className="btn title-button" onClick={this.postMenu}>
                    <p>Post Menu</p>
                  </button>
                  <button
                    className="title-button btn"
                    onClick={() => this.setState({ menu: [] })
                  }
                  >
                    <p>Clear Menu</p>
                  </button>
                </div>

                {
                  isMenuSet ?
                    <MealCardContainer
                      meals={this.state.menu}
                      MealCard={MealDisplayCard}
                      collection="Menu"
                      removeFremoveFromCollectionromMenu={this.removeFromMenu}
                      addClass="scroll"
                    /> :
                    <div className="menu-message">
                  `You have not added meals to the menu.
                   Select a meal from the meal section
                   above to add meal to menu.`
                    </div>
                 }
              </AccordionItemBody>
            </AccordionItem>

          </Accordion>
        </main>
      </div>
    );
  }
}
Menu.defaultProps = {
  menu: []
};
Menu.propTypes = {
  dispatch: PropTypes.func.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  menu: PropTypes.arrayOf(PropTypes.object),
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
    firstName: PropTypes.string
  }).isRequired,
};
const mapStateToProps = state => ({
  mealError: state.mealsReducer.mealError,
  connecting: state.mealsReducer.connecting,
  deleted: state.mealsReducer.deleted,
  meals: state.mealsReducer.meals,
  menu: state.menuReducer.menu.Meals,
  authenticated: state.loginReducer.authenticated,
  user: state.loginReducer.user.data
});

export { Menu };
export default connect(mapStateToProps)(Menu);
