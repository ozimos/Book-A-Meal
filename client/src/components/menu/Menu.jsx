import React from 'react';
import ReactModal from 'react-modal';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Accordion,
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from 'react-accessible-accordion';
import SearchInput, { createFilter } from 'react-search-input';
import MealCard2 from '../mealCard/MealCard2';
import MealCardContainer from '../mealCard/MealCardContainer';
import ConnectedGreeting from '../greeting/Greeting';
import { mealActions, menuActions } from '../../redux/actions';
import ConnectedNav from '../nav/Nav';
import '../../../public/styles/bookameal.scss';
import '../../../public/styles/modalOpenButton.scss';
import '../../../public/styles/search-input.css';
import '../../../public/styles/accordion.css';

ReactModal.setAppElement(document.getElementById('root'));
class Menu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      menu: []
    };
  }
  componentDidMount() {
    this.props.dispatch(mealActions.getAllMeals());
  }

  addToMenu = (meal) => {
    const inMenu = this.state.menu.some(elem => elem.id === meal.id);
    if (!inMenu) { this.setState(prevState => ({ menu: [...prevState.menu, meal] })); }
  }
  removeFromMenu = id =>
    this.setState(prevState =>
      ({ menu: prevState.menu.filter(elem => elem.id !== id) }));
  searchUpdated = (term) => {
    this.setState({ searchTerm: term });
  }
  postMenu = () => {
    const mealIdArray = this.state.menu.map(meal => meal.id);
    this.props.dispatch(menuActions.postMenu({ meals: mealIdArray }));
  }

  render() {
    const KEYS_TO_FILTERS = ['id', 'title', 'description', 'price'];

    const filteredMeals = this.props.meals
      .filter(createFilter(this.state.searchTerm, KEYS_TO_FILTERS));
    const { isCaterer, firstName } = this.props.user;
    return (
      <div className="contain">
        <header className="header">
          <ConnectedNav />
        </header>
        <main>
          <ConnectedGreeting isCaterer={isCaterer} firstName={firstName} />
          <Accordion accordion={false}>
            <AccordionItem expanded>
              <AccordionItemTitle>
                <div className="title-element flexbox">
                  <h4>
                  Your Meals
                  </h4>
                  <div className="accordion__arrow u-position-relative" role="presentation" />
                </div>
              </AccordionItemTitle>
              <AccordionItemBody>
                <SearchInput className="search-input input-field" onChange={this.searchUpdated} />
                <MealCardContainer
                  meals={filteredMeals}
                  MealCard={MealCard2}
                  addToMenu={this.addToMenu}
                  addClass="scroll"
                />
              </AccordionItemBody>
            </AccordionItem>
            <AccordionItem>
              <AccordionItemTitle>
                <div className="title-element flexbox wrap">
                  <h3>
          Today&#39;s Menu
                  </h3>
                  <div className="accordion__arrow u-position-relative" role="presentation" />
                </div>
              </AccordionItemTitle>
              <AccordionItemBody>
                <div className="title-element flexbox">
                  <button className="btn title-button" onClick={this.postMenu}>
                    <p>Post Menu</p>
                  </button>
                  <button className="title-button btn" onClick={() => this.setState({ menu: [] })}>
                    <p>Clear Menu</p>
                  </button>
                </div>

                <MealCardContainer
                  meals={this.state.menu}
                  MealCard={MealCard2}
                  removeFromMenu={this.removeFromMenu}
                  addClass="scroll"
                />
              </AccordionItemBody>
            </AccordionItem>

          </Accordion>
        </main>
      </div>
    );
  }
}
Menu.propTypes = {
  dispatch: PropTypes.func.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
    firstName: PropTypes.string
  }).isRequired,
};
const mapStateToProps = state => ({
  mealError: state.mealsReducer.mealError,
  connecting: state.mealsReducer.connecting,
  deleted: state.mealsReducer.deleted,
  meals: state.mealsReducer.meals,
  authenticated: state.loginReducer.authenticated,
  user: state.loginReducer.user.data
});

export { Menu };
export default connect(mapStateToProps)(Menu);
