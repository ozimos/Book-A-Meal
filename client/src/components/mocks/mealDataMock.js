export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
