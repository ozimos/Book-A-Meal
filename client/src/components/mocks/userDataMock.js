export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
