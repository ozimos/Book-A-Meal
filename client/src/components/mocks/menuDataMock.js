import { allMeals } from './mealDataMock';

export default {
  data: { rows: [{ Meals: allMeals.data.rows }] }
};
