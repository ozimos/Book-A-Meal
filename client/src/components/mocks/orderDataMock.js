import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};