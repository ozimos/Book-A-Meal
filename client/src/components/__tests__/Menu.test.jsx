/* global React:false, shallow:false toJson:false */

import { Menu } from '../menu/ConnectedMenu';
import { allMeals, meal } from '../mocks/mealDataMock';

const props = {
  menuPagination: {
    limit: 8,
    offset: 1,
    pages: 1
  },
  mealsPagination: {
    limit: 8,
    offset: 1,
    pages: 1
  },
  mealError: '',
  menuError: '',
  meals: allMeals.data.rows,
  menu: allMeals.data.rows,
  user: { isCaterer: true, firstName: 'user' },
  dispatch: jest.fn(),
};

describe('Menu Component', () => {
  const setup = () => shallow(<Menu {...props} />);


  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect((wrapper)
      .exists(<div className="title-element flexbox" />)).toBe(true);
  });
  it('renders correctly', () => {
    const menu = [];
    const menuPagination = {
      pages: 1
    };
    const mealsPagination = {
      pages: 1
    };
    const props2 = { ...props, menu };
    const props3 = { ...props, mealsPagination };
    const props4 = { ...props, menuPagination };
    let setup2 = () => shallow(<Menu {...props} />);

    let wrapper = setup2();
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.instance().setState({ currentMenu: allMeals.data.rows });
    expect(toJson(wrapper)).toMatchSnapshot();
    setup2 = () => shallow(<Menu {...props2} />);
    wrapper = setup2();
    expect(toJson(wrapper)).toMatchSnapshot();
    setup2 = () => shallow(<Menu {...props3} />);
    wrapper = setup2();
    expect(toJson(wrapper)).toMatchSnapshot();
    setup2 = () => shallow(<Menu {...props4} />);
    wrapper = setup2();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call `notify`', () => {
    const wrapper = setup();

    const notifySpy = jest.spyOn(
      wrapper.instance(),
      'notify'
    );
    wrapper.instance().notify();
    expect(notifySpy).toHaveBeenCalled();
  });
  it('should call `checkMeal`', () => {
    const wrapper = setup();
    wrapper.instance().setState({ currentMenu: allMeals.data.rows });
    const checkMealSpy = jest.spyOn(
      wrapper.instance(),
      'checkMeal'
    );
    wrapper.instance().checkMeal('abc');
    expect(checkMealSpy).toHaveBeenCalled();
  });
  it('should call `addToMenu`', () => {
    const wrapper = setup();
    wrapper.instance().setState({ currentMenu: allMeals.data.rows });

    const addToMenuSpy = jest.spyOn(
      wrapper.instance(),
      'addToMenu'
    );
    wrapper.instance().addToMenu(meal.data);
    expect(addToMenuSpy).toHaveBeenCalled();
  });
  it('should call `addToMenu` when no meals are selected', () => {
    const wrapper = setup();
    const addToMenuSpy = jest.spyOn(
      wrapper.instance(),
      'addToMenu'
    );
    wrapper.instance().addToMenu(meal.data);
    expect(addToMenuSpy).toHaveBeenCalled();
  });
  it('should call `openMenuModal`', () => {
    const wrapper = setup();

    const openMenuModalSpy = jest.spyOn(
      wrapper.instance(),
      'openMenuModal'
    );
    wrapper.instance().openMenuModal();
    expect(openMenuModalSpy).toHaveBeenCalled();
  });
  it('should call `closeMenuModal`', () => {
    const wrapper = setup();

    const closeMenuModalSpy = jest.spyOn(
      wrapper.instance(),
      'closeMenuModal'
    );
    wrapper.instance().closeMenuModal();
    expect(closeMenuModalSpy).toHaveBeenCalled();
  });
  it('should call `postMenu`', () => {
    const wrapper = setup();

    const postMenuSpy = jest.spyOn(
      wrapper.instance(),
      'postMenu'
    );
    wrapper.instance().postMenu();
    expect(postMenuSpy).toHaveBeenCalled();
  });
  it('should call `postMenu` when there is error', () => {
    const props2 = { ...props, menuError: 'abc' };
    const setup2 = () => shallow(<Menu {...props2} />);
    const wrapper = setup2();

    const postMenuSpy = jest.spyOn(
      wrapper.instance(),
      'postMenu'
    );
    wrapper.instance().postMenu();
    expect(postMenuSpy).toHaveBeenCalled();
  });
  it('should call `removeFromMenu`', () => {
    const wrapper = setup();

    const removeFromMenuSpy = jest.spyOn(
      wrapper.instance(),
      'removeFromMenu'
    );
    wrapper.instance().removeFromMenu('abc');
    expect(removeFromMenuSpy).toHaveBeenCalled();
  });
  it('should call `removeFromMenu` when there is no meal selected', () => {
    const wrapper = setup();
    wrapper.instance().setState({ currentMenu: [] });

    const removeFromMenuSpy = jest.spyOn(
      wrapper.instance(),
      'removeFromMenu'
    );
    wrapper.instance().removeFromMenu('abc');
    expect(removeFromMenuSpy).toHaveBeenCalled();
  });
  it('should call `removeFromMenu` when items are in menu', () => {
    const wrapper = setup();
    wrapper.instance().setState({ currentMenu: allMeals.data.rows });

    const removeFromMenuSpy = jest.spyOn(
      wrapper.instance(),
      'removeFromMenu'
    );
    wrapper.instance().removeFromMenu('abc');
    expect(removeFromMenuSpy).toHaveBeenCalled();
  });
  it('should call `handleMealPaginationClick`', () => {
    const wrapper = setup();

    const handleMealPaginationClickSpy = jest.spyOn(
      wrapper.instance(),
      'handleMealPaginationClick'
    );
    wrapper.instance().handleMealPaginationClick({ selected: 2 });
    expect(handleMealPaginationClickSpy).toHaveBeenCalled();
  });
  it('should call `clearMenu`', () => {
    const wrapper = setup();

    const clearMenuSpy = jest.spyOn(
      wrapper.instance(),
      'clearMenu'
    );
    wrapper.instance().clearMenu();
    expect(clearMenuSpy).toHaveBeenCalled();
  });
  it('should call `handleMenuPaginationClick`', () => {
    const wrapper = setup();

    const handleMenuPaginationClickSpy = jest.spyOn(
      wrapper.instance(),
      'handleMenuPaginationClick'
    );
    wrapper.instance().handleMenuPaginationClick({ selected: 2 });
    expect(handleMenuPaginationClickSpy).toHaveBeenCalled();
  });
});

