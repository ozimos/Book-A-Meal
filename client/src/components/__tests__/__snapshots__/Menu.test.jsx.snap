// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component renders correctly 1`] = `
<div
  className="contain"
>
  <header
    className="header"
  >
    <Connect(Nav) />
  </header>
  <main>
    <ToastContainer
      autoClose={2000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnHover={true}
      pauseOnVisibilityChange={true}
      position="top-right"
      progressClassName={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="flexbox"
    >
      <Greeting
        firstName="user"
        isCaterer={true}
      />
      <button
        className="btn title-button"
        disabled={true}
        onClick={[Function]}
      >
        <p
          className="cart-notification"
        >
          Menu
          <span
            className="badge"
          >
            0
          </span>
        </p>
      </button>
    </div>
    <AccordionWrapper
      accordion={true}
      className="accordion"
      onChange={[Function]}
    >
      <AccordionItemWrapper
        accordionStore={
          AccordionContainer {
            "_listeners": Array [],
            "state": Object {
              "accordion": true,
              "items": Array [],
              "onChange": [Function],
            },
          }
        }
        className="accordion__item"
        disabled={false}
        expanded={true}
        hideBodyClassName=""
      >
        <AccordionItemTitleWrapper
          className="accordion__title"
          hideBodyClassName=""
        >
          <div
            className="title-element flexbox"
          >
            <h5>
              Your Meals
            </h5>
            <div
              className="accordion__arrow u-position-relative"
              role="presentation"
            />
          </div>
        </AccordionItemTitleWrapper>
        <AccordionItemBodyWrapper
          className="accordion__body"
          hideBodyClassName="accordion__body--hidden"
        >
          <Search
            caseSensitive={false}
            className="search-input input-field"
            fuzzy={false}
            onChange={[Function]}
            throttle={200}
          />
          <MealCardContainer
            MealCard={[Function]}
            addClass="scroll"
            addToCollection={[Function]}
            collection="Menu"
            meals={Array []}
          />
          <PaginationBoxView
            activeClassName="active"
            breakLabel="..."
            containerClassName="pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={2}
            nextClassName="next"
            nextLabel="next"
            onPageChange={[Function]}
            pageCount={1}
            pageRangeDisplayed={5}
            previousClassName="previous"
            previousLabel="previous"
            subContainerClassName="pages pagination"
          />
        </AccordionItemBodyWrapper>
      </AccordionItemWrapper>
      <AccordionItemWrapper
        accordionStore={
          AccordionContainer {
            "_listeners": Array [],
            "state": Object {
              "accordion": true,
              "items": Array [],
              "onChange": [Function],
            },
          }
        }
        className="accordion__item"
        disabled={false}
        expanded={false}
        hideBodyClassName=""
      >
        <AccordionItemTitleWrapper
          className="accordion__title"
          hideBodyClassName=""
        >
          <div
            className="title-element flexbox wrap"
          >
            <h5>
              Today's Menu
            </h5>
            <div
              className="accordion__arrow u-position-relative"
              role="presentation"
            />
          </div>
        </AccordionItemTitleWrapper>
        <AccordionItemBodyWrapper
          className="accordion__body"
          hideBodyClassName="accordion__body--hidden"
        >
          <div
            className="title-element flexbox"
          >
            <button
              className="title-button btn"
              onClick={[Function]}
            >
              <p>
                Clear Menu
              </p>
            </button>
          </div>
          <div
            className="menu-message"
          >
            You have not added meals to the menu. Select a meal from the meal section above to add meal to menu.
          </div>
        </AccordionItemBodyWrapper>
      </AccordionItemWrapper>
    </AccordionWrapper>
  </main>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-content"
    closeTimeoutMS={0}
    contentLabel="Input Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <aside
      className="col s12"
    >
      <div>
        <h3>
          Menu
        </h3>
        <p>
          No meals here. Select a meal and click the Add to Menu button
        </p>
      </div>
    </aside>
  </Modal>
</div>
`;
